subprojects/core-api/src/main/java/org/gradle/api/tasks/CompileClasspath.java

```diff
@@ -35,6 +35,14 @@
  *     <li>Changes to directories, including directory entries in Jars.</li>
  * </ul>
  *
+ * <p>
+ *     When used on an {@literal @}{@link org.gradle.work.Incremental} input, Gradle will not produce any
+ *     {@link org.gradle.work.ChangeType#MODIFIED} events.
```

Comment:
❌ This is not true. If possible, we try to identify elements on classpath inputs via the absolute path to produce modified events. So if, for example, a project JAR changes, we report it as `MODIFIED`. For external dependencies, that doesn't work, though.

See for example this test: https://github.com/gradle/gradle/blob/9bd595ea31b56b19fd2487c928dd37b201c3eb05/subprojects/execution/src/test/groovy/org/gradle/internal/execution/history/changes/ClasspathCompareStrategyTest.groovy#L72-L78